homeassistant/components/incomfort/config_flow.py

```diff
@@ -102,6 +119,65 @@ def async_get_options_flow(
         """Get the options flow for this handler."""
         return InComfortOptionsFlowHandler()
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a DHCP discovered Intergas Gateway device."""
+        self._discovered_host = discovery_info.ip
+        # In case we have an existing entry with the same host
+        # we update the entry with the unique_id for the gateway, and abort the flow
+        unique_id = format_mac(discovery_info.macaddress)
+        existing_entries = [
```

Comment:
Maybe name it `existing_entries_without_unique_id`

Reply:
Still that not so great, but it took me a minute it was looking up the entry for the existing host without a unique id so it could set it